pkg/scheduler/framework/types.go

```diff
@@ -378,8 +378,15 @@ func getNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffini
 type ImageStateSummary struct {
 	// Size of the image
 	Size int64
-	// Used to track how many nodes have this image
-	NumNodes int
+	// A set of node names for nodes having this image present
+	Nodes sets.String
```

Comment:
Actually, let's keep track of both the set and the number of nodes in this object.

Then the function `ImageStateSummary.Clone` below can be renamed to `Snapshot` and only copy the number, but not the set, because that's all we need during a snapshot.